### PR TITLE
fix:[2.4]Chunk Id out of range in vector BF search, after the growing index removes the vec chunks.

### PR DESCRIPTION
**Pull request by cqy123456**
issue: https://github.com/milvus-io/milvus/issues/36871
related pr: https://github.com/milvus-io/milvus/pull/36939

### CODE REVIEW

File: internal/core/src/query/SearchOnGrowing.cpp [hunk 1]
```diff
@@ -94,15 +94,20 @@ SearchOnGrowing(const segcore::SegmentGrowingImpl& segment,
         FloatSegmentIndexSearch(
             segment, info, query_data, num_queries, bitset, search_result);
     } else {
+        std::shared_lock<std::shared_mutex> read_chunk_mutex(
+            segment.get_chunk_mutex());
+        // check SyncDataWithIndex() again, in case the vector chunks has been removed.
+        if (segment.get_indexing_record().SyncDataWithIndex(field.get_id())) {
+            FloatSegmentIndexSearch(
```
Comment (foxspy) [inline internal/core/src/query/SearchOnGrowing.cpp:101]: return here? it seems that no need to run the following broute force

Comment (cqy123456) [reply internal/core/src/query/SearchOnGrowing.cpp]: updated

**Pull request merged**